### DISCUSSION TITLE
Looking forward

### DISCUSSION THREAD

**Discussion by CAD97** [pest-parser/pest#547] [General]
I didn't really want to have to write this, but...

It seems @dragostis is no longer available to work on Pest. The most recent interaction Dragoș had with the repo is a single one in [March 2020](https://github.com/pest-parser/pest/commit/7dee2a7770daa213b45f88e6af730868f764927a), and the ones before that was [December 2018](https://github.com/pest-parser/pest/commits?author=dragostis&since=2018-12-01&until=2019-01-01). Additionally, Dragoș [hasn't been very active on GitHub recently at all](https://github.com/dragostis).

Dragoș is the only user with permissions to publish the pest package to crates-io. However, I do have admin permissions in the @pest-parser org, so we still have control over the GitHub side.

On a personal side, I'm both currently pursuing a rewarding path into game development (currently as a graduate student), and have fallen out of favor with the lexerless style that Pest uses. As such, I don't have much time nor drive to work on Pest; I'm perfectly happy to serve as a reviewer and steward, but won't be putting in any development effort.

So this is an open question for the users: how do you want to proceed? This is the community's project now!

**Comment by NoahTheDuke** [comment]
I'm new to Pest but am interested in contributing. It seems that without the ability to publish to crates.io, we're at a standstill even if a new maintainer is found. Is the next step a fork and rename?

**Reply by CAD97** [reply]
That seems to be the case. I got a response to an email to @dragostis about the pest.rs domain, so I'll send another email sometime soon asking if he can add me as a publisher on crates-io, but otherwise, a new crates-io package name would be required.

**Comment by Geobert** [comment]
There is this that could help with the process: https://users.rust-lang.org/t/bus-factor-1-for-crates/17046

**Comment by jonasvanderschaaf** [comment]
This seems like a really cool project and I'd love to contribute to it. I am rather busy the next two months, but after that I could help out with the project if necessary.

**Comment by SaadiSave** [comment]
Pest has been very helpful for [me](https://github.com/SaadiSave/cambridge-asm). I would love to contribute if possible. I am busy until May '22, but I can contribute after.

P.S. Please provide an update about the crates.io situation soon.

**Comment by salmmanfred** [comment]
id love to contribute

**Comment by dragostis** [comment]
Hey everyone! I'm happy to provide access to crates.io when a new maintainer is found. Sorry for not being more active lately, I've just been quite severely affected by the pandemic and all the time spent alone and inside which made it hard to find the motivation to work on anything outside of work.

That being said, I'd be very happy to mentor someone on pest3 if anyone is interested on doing core work and learning about low-level, high-efficiency computing. Although I haven't posted any updates there are a few ideas I've experimented with:

* new grammar is already implemented in pest2 and it's much clearer and more usable
* a much better procedural macro that actually works like a real parser where you can match on exactly the right cases based on the grammar you define
* an MIR for pest3 combined with [egg](https://docs.rs/egg/0.6.0/egg/) that will hopefully remove all exponential edge cases
* a new code generator that reduces the MIR to a state machine connected to one another by simple Rust matches

**Reply by CAD97** [reply]
I'd suggest adding the @pest-parser/admins team as a publisher for the pest crates; that'll allow us to manage trusted publishers in one central location. (And as a plus, that already includes me, so we're no longer bottlenecked on just you for publishing.)

I'm glad you're doing well enough ❤️ The pandemic's been rough on all of us, but some more than others.

**Reply by dragostis** [reply]
How can I add the admins team on crates.io?

**Reply by CAD97** [reply]
`cargo owner`: https://doc.rust-lang.org/cargo/reference/publishing.html#cargo-owner

**Reply by dragostis** [reply]
Thank you. I've added the admins group. :)

**Comment by jhoobergs** [comment]
I really like using pest to create parsers, so I would like to help. I am mainly interested in systems like https://github.com/pest-parser/ast. I seem to be writing way to much code to consume the pest result and convert it to an ast.

**Reply by dragostis** [reply]
This is what pest3 would be about. The basic idea is to use the grammar like `pair = { integer ~ ":" ~ string }` to parse directly into either `#[rule] fn pair(i: i32, s: &str) -> ...` or `#[rule(pair)] struct Pair { i: i32, s: String }`.

**Reply by jhoobergs** [reply]
Great!

**Reply by jhoobergs** [reply]
How would `integer` and `string` be defined in this grammar?  I mean, how would we know that `integer` maps to `i32` and `string` to `String`.?
